---
title: "Compiler Error CS0103 | Microsoft Docs"
ms.date: "2015-07-20"
ms.prod: ".net"
ms.technology: 
  - "devlang-csharp"
ms.topic: "error-reference"
f1_keywords: 
  - "CS0103"
dev_langs: 
  - "CSharp"
helpviewer_keywords: 
  - "CS0103"
ms.assetid: fd1f2104-a945-4dba-8137-8ef869826062
caps.latest.revision: 13
author: "BillWagner"
ms.author: "wiwagn"
caps.handback.revision: 13
---
# Compiler Error CS0103
Il nome 'identificatore' non esiste nel contesto corrente.  
  
 È stato effettuato un tentativo di utilizzare un nome non presente nella classe, nello [spazio dei nomi](../../../csharp/language-reference/keywords/namespace.md) o nell'ambito.  Controllare l'ortografia del nome, le direttive in uso e i riferimenti all'assembly per verificare la disponibilità del nome che si tenta di utilizzare.  
  
 Questo errore si verifica frequentemente quando si dichiara una variabile all'interno di un ciclo o di un blocco `try` o `if` e successivamente si tenta di accedere alla variabile dal blocco di codice che la contiene o da un altro blocco di codice, come mostrato nell'esempio che segue.  
  
```c#  
using System;  
  
class MyClass1  
{  
    public static void Main()  
    {  
        try  
        {  
            // The following declaration is only available inside the try block.  
            MyClass1 conn = new MyClass1();  
        }  
        catch (Exception e)  
        {  
            // The following expression causes error CS0103, because variable  
            // conn only exists in the try block.  
            if (conn != null)   
                Console.WriteLine("{0}", e);  
        }  
    }  
}  
  
```  
  
 L'esempio seguente risolve il problema.  
  
```c#  
  
using System;  
  
class MyClass2  
{  
    public static void Main()  
    {  
        // To resolve the error in the example, the first step is to   
        // move the declaration of conn out of the try block. The following  
        // declaration is available throughout the Main method.  
        MyClass2 conn = null;  
        try  
        {  
            // Inside the try block, use the conn variable that you declared  
            // previously.  
            conn = new MyClass2();  
        }  
        catch (Exception e)  
        {  
            // The following expression no longer causes an error, because  
            // the declaration of conn is in scope.  
            if (conn != null)   
                Console.WriteLine("{0}", e);  
        }  
    }  
}  
  
```